client/crypto.go

```diff
@@ -0,0 +1,79 @@
+package client
```

Comment:
crypto part should be reusable by server and the client and the rest of the world, let's move it under `pkg`